packages/mobile/src/tokens/reducer.ts

```diff
@@ -0,0 +1,45 @@
+import { RehydrateAction } from 'redux-persist'
+import { getRehydratePayload, REHYDRATE } from 'src/redux/persist-helper'
+import { Actions, ActionTypes } from 'src/tokens/actions'
+
+export interface Token {
+  address: string
+  decimals: number
+  imageUrl: string
+  name: string
+  symbol: string
+}
+
+export interface TokenBalance extends Token {
+  balance: number
+}
+
+export interface TokenBalances {
+  [address: string]: TokenBalance
+}
+
+export interface State {
+  tokenBalances: TokenBalances
+}
+
+export const initialState = {
+  tokenBalances: {},
+}
+
+export const reducer = (state: State = initialState, action: ActionTypes | RehydrateAction) => {
```

Comment:
One consideration is that we're retying to move towards [redux toolkit](https://redux-toolkit.js.org/) which is less verboragic about creating all these things. One example is the recipients reducer.

No need to change it now, but I might if I get some extra time here or there if you don't :)